packages/pigeon/lib/swift_generator.dart

```diff
@@ -677,8 +678,8 @@ String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
   } else if (type.baseName == 'Object') {
     // Special-cased to avoid warnings about using 'as' with Any.
     return value;
-  } else if (type.baseName == 'int') {
-    return '($value is Int) ? Int64($value as! Int) : $value as! Int64$castUnwrap';
+  } else if (type.isNullable) {
+    return '($value is NSNull) ? (nil as Any?) as! ${_swiftTypeForDartType(type)}$castUnwrap : $value as! ${_swiftTypeForDartType(type)}$castUnwrap';
```

Comment:
I think we should make a generated private helper function that converts NSNull to nil rather than adding this logic everywhere. Similar to the `GetNullableObjectAtIndex` we have for Obj-C, but it can just take a variable instead of an array and index.

Reply:
I had that originally, and it ended up just being a messier version of this. I can revert it to that though.

Reply:
Generics in swift are really strange as you can't specify what the generic type is at run time. It has to be infered and there is a lot of weirdness with that?